### PR TITLE
Update release guide for download page changes

### PR DESCRIPTION
**Pull request by boyuanzz**
r: @pabloem

### CODE REVIEW

File: src/contribute/release-guide.md [hunk 1]
```diff
@@ -778,6 +778,8 @@ Note: It is important to rename `apache-beam-${RELEASE}-python.zip` to `apache-b
 
 Copy the source release from the `dev` repository to the `release` repository at `dist.apache.org` using Subversion.
 
+Move last release artifacts from `dist.apache.org` to `archive.apache.org` using Subversion, [example PR](https://github.com/apache/beam-site/pull/478).
```
Comment (pabloem) [inline src/contribute/release-guide.md]: It seems that the PR that you're linking is only a change in documentation, and not a good example of how to move the sources into the archive?

Comment (boyuanzz) [reply src/contribute/release-guide.md]: Added update beam-site in commit https://github.com/apache/beam-site/pull/486/commits/d6d9627f1489db37b616f1cca7aacb0fcd09c534

**Comment by pabloem** [comment]
LGTM.

**Comment by pabloem** [comment]
@asfgit merge

**Pull request merged**